src/components/CharacterForm/CharacterForm.js

```diff
@@ -98,6 +99,47 @@ const CharacterForm = ({ setOpen }) => {
     )
   }
 
+  const submit = handleSubmit(onSubmit, (errors) => {
+    setErrors(errors)
+  })
+
+  const rules = {
+    name: {
+      required: "This field is required",
+      minLength: {
+        value: 3,
+        message: "This field should contain at least 3 characters",
+      },
+      pattern: {
+        value: /^[^\s]{3}.*/,
+        message: "The first 3 characters shouldn't be space!",
+      },
+    },
+    eyeColor: {
+      required: "This field is required",
+    },
+    height: {
+      min: {
+        value: 1,
+        message: "The height should range from 1 - 300",
+      },
+      max: {
+        value: 300,
+        message: "The height should range from 1 - 300",
+      },
+    },
+    homeworld: {
+      required: "This field is required",
```

Comment:
This "required" rule seems to be quite popular and we might also want to use it again in other forms, how can you make this re-usable so so don't have to declare this rule in every form that uses the inputs?